### PR TITLE
[OTA] add register device command

### PR DESCRIPTION
**Pull request by alexischr**

### CODE REVIEW

File: Meadow.CLI.Core/Devices/MeadowLocalDevice.cs [hunk 1]
```diff
@@ -364,6 +364,22 @@ public async Task SetRtcTime(DateTimeOffset dateTime, CancellationToken cancella
...
+
+            var command =
+                new SimpleCommandBuilder(HcomMeadowRequestType.HCOM_MDOW_REQUEST_OTA_REGISTER_DEVICE)
+                    .WithResponseType(MeadowMessageType.Data)
+                    .Build();
+
+            var commandResponse =
+                await SendCommand(command, cancellationToken);
+
+            Console.WriteLine($"Device's public key: {commandResponse.Message}");
```
Comment (adrianstevens) [inline Meadow.CLI.Core/Devices/MeadowLocalDevice.cs:379]: @alexischr we can't use Console.WriteLine in the CLI - can you switch to the logger?

File: Meadow.CLI/Commands/Cloud/RegisterDeviceCommand.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        public CloudRegisterCommand(DownloadManager downloadManager, ILoggerFactory loggerFactory, MeadowDeviceManager meadowDeviceManager)
+            : base(downloadManager, loggerFactory, meadowDeviceManager)
+        {
+        }
+
+        public async ValueTask ExecuteAsync(IConsole console)
+        {
+            var cancellationToken = console.RegisterCancellationHandler();
+
+            // var identityManager = new IdentityManager();
```
Comment (adrianstevens) [inline Meadow.CLI/Commands/Cloud/RegisterDeviceCommand.cs:27]: Looks like this is still draft - so just a friendly note to clean up any unused code :)

File: Meadow.CLI.Core/Devices/MeadowLocalDevice.cs (hunk 1 above)
Comment (alexischr) [reply Meadow.CLI.Core/Devices/MeadowLocalDevice.cs:379]: Yup, this is a work in progress with @lamebrain and I replacing the console output and disabled code with a push of the data to Meadow Cloud

**Comment by alexischr** [comment]
Currently this new command returns right away, seemingly not sending the request. Investigating…

**Comment by alexischr** [comment]
Above issue fixed (was copy/pasting seemingly bad code lacking an `override`)

Now the command times out waiting for the operation to conclude (RSA keygen takes a while on the STM32F7)

File: Meadow.CLI/Meadow.CLI.csproj [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-	<TargetFrameworks>net5.0;net6.0</TargetFrameworks>
+	<TargetFrameworks>net5.0</TargetFrameworks>
```
Comment (adrianstevens) [inline Meadow.CLI/Meadow.CLI.csproj:5]: We'll need to discuss .NET versions, the last beta release is .NET 6.0 only and with .NET 5.0 being out of support we should probably move away from it.

File: Meadow.CLI.Core/Devices/MeadowLocalDevice.cs (hunk 1 above)
Comment (lamebrain) [reply Meadow.CLI.Core/Devices/MeadowLocalDevice.cs]: meadow cloud portion added now

**Comment by CartBlanche** [comment]
Looks like quite a few of the commits are already in `develop` . Could a rebase be missing?

**Pull request merged**